code/modules/hydroponics/hydro_tray_process.dm

```diff
@@ -215,7 +215,7 @@
 	set_light(light_out)
 
 	var/light_available = 5
-	if(T?.dynamic_lighting)
+	if(istype(T) && T.dynamic_lighting)
```

Comment:
Why is `T` ever not a turf?

Reply:
I think plants that emit light locked in closets might cause the issue, so I think this resolves it

Reply:
`var/turf/T = loc`